### PR TITLE
[fix](storage) storage layer should aggregate on replace column

### PR DESCRIPTION
**Pull request by dataroaring**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataalive — APPROVED** [review]
LGTM

File: be/src/exec/olap_scanner.cpp [hunk 1]
```diff
@@ -152,14 +152,15 @@ Status OlapScanner::_init_tablet_reader_params(
                       ->rowset()
                       ->rowset_meta()
                       ->is_segments_overlapping());
+    
+    bool has_replace_col = false;
+    RETURN_IF_ERROR(_init_return_columns(!single_non_overlapping_version, has_replace_col));
 
-    _tablet_reader_params.direct_mode = single_version || _aggregation;
```
Comment (zhannngchen) [inline be/src/exec/olap_scanner.cpp:156]: The issue here is:
1. Planner thinks scanner can scan directly, don't need merge and can do some pre-aggregation works.
2. But scanner found it's a unqiue key table with seq column which need to merge first.

I think we'd fix it in planner? Planner should not set preAggregation as true on a unique table with seq column?

Comment (zhannngchen) [inline be/src/exec/olap_scanner.cpp:156]: Is the following change also can fix the issue?
```
bool unique_key_with_seq =  _tablet->keys_type() == UNIQUE_KEYS && _tablet->tablet_schema().has_sequence_col();
_tablet_reader_params.direct_mode = single_version || (_aggregation && ! unique_key_with_seq) ;
```

Comment (dataroaring) [reply be/src/exec/olap_scanner.cpp:156]: I am not sure what we need to in planner.  If fe sets aggregation, we ignore it here, then we can ignore aggreagate in planner somtimes?  Could you throw an example sql?

Comment (dataroaring) [reply be/src/exec/olap_scanner.cpp:156]: I  suspect that columns aggregated by replace without SEQ does not work either.

Comment (zhannngchen) [reply be/src/exec/olap_scanner.cpp:156]: > I suspect that columns aggregated by replace without SEQ does not work either.

root cause is _aggregation should not be true in such situation?

**Pull request closed without merging**